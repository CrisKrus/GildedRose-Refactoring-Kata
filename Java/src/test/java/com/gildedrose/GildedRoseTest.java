package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void updateItemQuality() {
        CombinationApprovals.verifyAllCombinations(
            (name, sellIn, quality) -> updateQuality(name, sellIn, quality).toString().trim(),
            new String[]{"foo", "Aged Brie", "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert"}, // name
            new Integer[]{0, 5, 10, 100}, // sellIn
            new Integer[]{0, 5, 10, 50} // quality
        );
    }

    private static Item updateQuality(String name, int sellIn, int quality) {
        Item item = new Item(name, sellIn, quality);
        GildedRose app = new GildedRose(new Item[] { item });
        app.updateQuality();
        return item;
    }

}
