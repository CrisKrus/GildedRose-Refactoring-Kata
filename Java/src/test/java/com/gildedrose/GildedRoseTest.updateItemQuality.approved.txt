[foo, 0, 0] => foo, -1, 0
[foo, 0, 5] => foo, -1, 3
[foo, 0, 10] => foo, -1, 8
[foo, 0, 50] => foo, -1, 48
[foo, 5, 0] => foo, 4, 0
[foo, 5, 5] => foo, 4, 4
[foo, 5, 10] => foo, 4, 9
[foo, 5, 50] => foo, 4, 49
[foo, 10, 0] => foo, 9, 0
[foo, 10, 5] => foo, 9, 4
[foo, 10, 10] => foo, 9, 9
[foo, 10, 50] => foo, 9, 49
[foo, 100, 0] => foo, 99, 0
[foo, 100, 5] => foo, 99, 4
[foo, 100, 10] => foo, 99, 9
[foo, 100, 50] => foo, 99, 49
[Aged Brie, 0, 0] => Aged Brie, -1, 2
[Aged Brie, 0, 5] => Aged Brie, -1, 7
[Aged Brie, 0, 10] => Aged Brie, -1, 12
[Aged Brie, 0, 50] => Aged Brie, -1, 50
[Aged Brie, 5, 0] => Aged Brie, 4, 1
[Aged Brie, 5, 5] => Aged Brie, 4, 6
[Aged Brie, 5, 10] => Aged Brie, 4, 11
[Aged Brie, 5, 50] => Aged Brie, 4, 50
[Aged Brie, 10, 0] => Aged Brie, 9, 1
[Aged Brie, 10, 5] => Aged Brie, 9, 6
[Aged Brie, 10, 10] => Aged Brie, 9, 11
[Aged Brie, 10, 50] => Aged Brie, 9, 50
[Aged Brie, 100, 0] => Aged Brie, 99, 1
[Aged Brie, 100, 5] => Aged Brie, 99, 6
[Aged Brie, 100, 10] => Aged Brie, 99, 11
[Aged Brie, 100, 50] => Aged Brie, 99, 50
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros, 5, 0] => Sulfuras, Hand of Ragnaros, 5, 0
[Sulfuras, Hand of Ragnaros, 5, 5] => Sulfuras, Hand of Ragnaros, 5, 5
[Sulfuras, Hand of Ragnaros, 5, 10] => Sulfuras, Hand of Ragnaros, 5, 10
[Sulfuras, Hand of Ragnaros, 5, 50] => Sulfuras, Hand of Ragnaros, 5, 50
[Sulfuras, Hand of Ragnaros, 10, 0] => Sulfuras, Hand of Ragnaros, 10, 0
[Sulfuras, Hand of Ragnaros, 10, 5] => Sulfuras, Hand of Ragnaros, 10, 5
[Sulfuras, Hand of Ragnaros, 10, 10] => Sulfuras, Hand of Ragnaros, 10, 10
[Sulfuras, Hand of Ragnaros, 10, 50] => Sulfuras, Hand of Ragnaros, 10, 50
[Sulfuras, Hand of Ragnaros, 100, 0] => Sulfuras, Hand of Ragnaros, 100, 0
[Sulfuras, Hand of Ragnaros, 100, 5] => Sulfuras, Hand of Ragnaros, 100, 5
[Sulfuras, Hand of Ragnaros, 100, 10] => Sulfuras, Hand of Ragnaros, 100, 10
[Sulfuras, Hand of Ragnaros, 100, 50] => Sulfuras, Hand of Ragnaros, 100, 50
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 10] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert, 5, 0] => Backstage passes to a TAFKAL80ETC concert, 4, 3
[Backstage passes to a TAFKAL80ETC concert, 5, 5] => Backstage passes to a TAFKAL80ETC concert, 4, 8
[Backstage passes to a TAFKAL80ETC concert, 5, 10] => Backstage passes to a TAFKAL80ETC concert, 4, 13
[Backstage passes to a TAFKAL80ETC concert, 5, 50] => Backstage passes to a TAFKAL80ETC concert, 4, 50
[Backstage passes to a TAFKAL80ETC concert, 10, 0] => Backstage passes to a TAFKAL80ETC concert, 9, 2
[Backstage passes to a TAFKAL80ETC concert, 10, 5] => Backstage passes to a TAFKAL80ETC concert, 9, 7
[Backstage passes to a TAFKAL80ETC concert, 10, 10] => Backstage passes to a TAFKAL80ETC concert, 9, 12
[Backstage passes to a TAFKAL80ETC concert, 10, 50] => Backstage passes to a TAFKAL80ETC concert, 9, 50
[Backstage passes to a TAFKAL80ETC concert, 100, 0] => Backstage passes to a TAFKAL80ETC concert, 99, 1
[Backstage passes to a TAFKAL80ETC concert, 100, 5] => Backstage passes to a TAFKAL80ETC concert, 99, 6
[Backstage passes to a TAFKAL80ETC concert, 100, 10] => Backstage passes to a TAFKAL80ETC concert, 99, 11
[Backstage passes to a TAFKAL80ETC concert, 100, 50] => Backstage passes to a TAFKAL80ETC concert, 99, 50
